processor/servicegraphprocessor/processor.go

```diff
@@ -166,7 +166,7 @@ func (p *serviceGraphProcessor) ConsumeTraces(ctx context.Context, td ptrace.Tra
 
 	// Skip empty metrics.
 	if md.MetricCount() == 0 {
-		return nil
+		return p.tracesConsumer.ConsumeTraces(ctx, td)
```

Comment:
It should still do the check from line 173

```
if p.tracesConsumer == nil {
}
```

Reply:
done